include/macros.h

```diff
@@ -21,8 +24,8 @@
 #define ACTIVE_CAM globalCtx->cameraPtrs[globalCtx->activeCamera]
 
 #define SET_NEXT_GAMESTATE(curState, newInit, newStruct) \
-            (curState)->nextGameStateInit = (GameStateFunc)newInit; \
-            (curState)->nextGameStateSize = sizeof(newStruct);
+            (curState)->init = (GameStateFunc)newInit; \
```

Comment:
I think you should keep the names for `nextGameStateInit` and `nextGameStateSize`. I think just calling them `init` and `size` is misleading, and it should be clear that they refer only to the _next_ game state

Reply:
But don't they point to whatever the init function and size of the current game state is? I mean it could be seen as next game state because if you set state->init = func itll use func as the current init function if im not mistaken. I could be wrong though. 

I think I was picturing it like this. If I was to look at the state in ram, lets say the current state is file context. so state->init would equal the file context init function, right? And then if you wanted to switch to the gameplay context, you could set the current init function to the gameplay init function. 

I could be completely wrong on this, its just what I thought and I can't check it in actual ram right now because I'm on my school computer lol

Reply:
@shawlucas Sorry for the late reply. The way that changing the gamestates work is that you set `nextGameStateInit` and `nextGameStateSize`, then set `isRunning` to 0. Then `GameState_Init` is called, which actually sets `nextGameStateInit` and `nextGameStateSize` both to 0. I think if you want know what the gamestate while the game is running, you'd look at the `main` function pointer.